### DISCUSSION TITLE
Feat: rotate through themes

### DISCUSSION THREAD

**Discussion by sethgrid** [ghostty-org/ghostty#3577] [Feature Requests, Ideas] [answered]
on each new window or on each call to cmd+shift+, - shuffle or use the next theme in the theme config. update the theme config to take an optional list of themes.

each time you open the terminal to the joy and surprise of a bit of visual variety

**Comment by 00-kat** [comment, marked as answer]
You may be interested in a workaround someone in the [Ghostty Discord server](https://discord.com/ghostty) found: https://discord.com/channels/1005603569187160125/1320117207405170813/1320144666557157426.  The trick is to run a script that modifies the Ghostty config using the `command` config entry.

**Reply by sethgrid** [reply]
this works. Just crafted up a function in zshrc that writes a random favorite template to the linked config file. every new tab, if I want, I reload config to get visual variety

**Comment by merinids212** [comment]
late to this but I ended up going a different route — OSC escape sequences from zshrc instead of rewriting the config file. each tab gets its own random theme, nothing on disk changes.

been using it daily for a while, finally cleaned it up: https://github.com/merinids212/ghostty-random-theme

**Reply by 8bittts** [reply]
that's actually looks pretty fun!

**Reply by adam-stone-zocdoc** [reply]
this is awesome!

**Comment by flyerAI2025** [comment]
For anyone looking for a workaround today — I made **[ghostty-themes](https://github.com/flyerAI2025/ghostty-themes)**, a single zsh script that lets you browse all 300+ themes with fzf and live preview. The theme applies to your terminal in real-time as you navigate — so you can quickly flip through them, find what you like, and press Enter to confirm.

More details in my [comment on #8145](https://github.com/ghostty-org/ghostty/discussions/8145#discussioncomment-16033422).
